### PR TITLE
Add Source Applications base page

### PR DESCRIPTION
**Pull request by igneel64**

### CODE REVIEW

File: docs/understanding-tracking-design/defining-the-data-to-collect-with-data-poducts/index.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ With data products, you can:
 - **Description;** a description of the data that the data product captures
 - **Owner;** the individual responsible for the data product 
 - **Domain;** the team or business domain that owns the data product
+- **Source Application;** the [source application/s](../organize-data-sources-with-source-applications/index.md) the Data Product is implemented in
```
Comment (gibbok-snowplow) [inline docs/understanding-tracking-design/defining-the-data-to-collect-with-data-poducts/index.md:26]: please consider this version, IMO a bit more clear

"Source Application in which the Data Product is implemented"

File: docs/understanding-tracking-design/organize-data-sources-with-source-applications/index.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+title: "Organize Data Sources with Source Applications"
+sidebar_position: 1
+sidebar_label: "🆕 Source Applications"
+---
+
+For data collection, you will often have different sources of information that correspond to applications designed for a particular purpose. That is what we will be referring to as Source Applications.
```
Comment (gibbok-snowplow) [inline docs/understanding-tracking-design/organize-data-sources-with-source-applications/index.md:7]: slightly revised version WDYT?:

```
For data collection, you will often encounter different sources of information that correspond to applications designed for specific purposes. These are what we will refer to as Source Applications.

To illustrate, let's consider Snowplow. We can identify several applications designed for distinct purposes, each serving as a separate data source for behavioral data—or, in other words, a Source Application:

- The Snowplow website, corresponding to the application served at www.snowplow.io.
- The BDP Console application, served at console.snowplowanalytics.com.
- The documentation website, which serves as our information hub for all things related to our product, available at docs.snowplow.io.

Source Applications are a foundational component that enables you to establish the overarching relationships that connect application IDs and [Application Entities].

``` 
on the right the new version

Screenshot 2024-08-14 at 9 58 02 AM

File: docs/understanding-tracking-design/organize-data-sources-with-source-applications/index.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+- The Snowplow website that corresponds to the application served under www.snowplow.io
+- The BDP Console application that is served under console.snowplowanalytics.com.
+- The documentation website serving as our information hub, for all things related to our product, served under docs.snowplow.io.
+
+Source Applications are a foundational component that will allow you to create the overarching relationship connecting application IDs, [Application Entites](../../collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/custom-tracking-using-schemas/global-context/index.md) and [Data Products](../defining-the-data-to-collect-with-data-poducts/index.md).
+
+## Application IDs
+
+For each of these applications you would set up a unique application ID using the [app_id](../../collecting-data/collecting-from-own-applications/snowplow-tracker-protocol/ootb-data/app-information/index.md#atomic-event-properties) field to discern them later on in analysis.
```
Comment (gibbok-snowplow) [inline docs/understanding-tracking-design/organize-data-sources-with-source-applications/index.md:18]: in stead of discern I would use a more simpler word like:

```
This allows you to distinguish between them during analysis.
```

File: docs/understanding-tracking-design/organize-data-sources-with-source-applications/index.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+:::tip
+What we often see and consider a best practice is to also set up an application ID for each deployment environment you are utilizing. For example `${appId}-qa` for staging, `${appId}-dev` for development environments.
```
Comment (gibbok-snowplow) [inline docs/understanding-tracking-design/organize-data-sources-with-source-applications/index.md:21]: We often see, and recommend as a best practice, setting up a unique application ID for each deployment environment you are using.

File: docs/understanding-tracking-design/organize-data-sources-with-source-applications/index.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+:::
+
+## Application Context
+
+Application Context, also referred to as [Global Context](../../collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/custom-tracking-using-schemas/global-context/index.md), is a set of entities that can be sent with every event recorded on the application. Using Source Applications you can document which Application Contexts are expected. This is really useful for tracking implementation, data discovery and preventing information duplication in Data Products.
```
Comment (gibbok-snowplow) [inline docs/understanding-tracking-design/organize-data-sources-with-source-applications/index.md:26]: on the application. => in the application.

File: docs/understanding-tracking-design/organize-data-sources-with-source-applications/index.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+:::info
+Since Application Entities can also be set conditionally, you can mark any of them as optional with a note to better understand the condition or any extra information required. The ways to conditionally add an Application Context is through [rulesets](../../collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/custom-tracking-using-schemas/global-context/index.md#rulesets), [filter functions](../../collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/custom-tracking-using-schemas/global-context/index.md#filter-functions) and [context generators](../../collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/custom-tracking-using-schemas/global-context/index.md#context-generators).
```
Comment (gibbok-snowplow) [inline docs/understanding-tracking-design/organize-data-sources-with-source-applications/index.md:29]: "The ways to conditionally add an Application Context is" => "The method for conditionally adding an Application Context is"

**Review by jamessnowplow — APPROVED** [review]
Looks good. Do we also need to go through and update the data product docs anywhere?

**Comment by jamessnowplow** [comment]
> As far as I can see we don't mention app Ids

Not for now, but there are probably a few words we could add to the data products docs to explain things a bit and point back here.

**Pull request merged**